File: LaserPolishingModel/Parameter/Laser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LaserPolishingModel.Parameter
{
    public class Laser : IParameterCollection
    {
        public Parameter BeamRadius
        {
            get; set;
        }

        public Parameter PulseDuration
        {
            get; set;
        }

        public Parameter AveragePower
        {
            get; set;
        }

        public Parameter DutyCycle
        {
            get; set;
        }

        public Parameter PulseFrequency
        {
            get; set;
        }

        // computed from others //
        // TODO: make this more elegant - not hardcoded.... //
        public Parameter PulseAveragePower
        {
            get
            {
                return new Parameter(
                    "Pulse Average Power",
                    AveragePower.Value / (PulseFrequency.Value * PulseDuration.Value));
            }
        }

        public IEnumerable<Parameter> GetParameters()
        {
            return new List<Parameter>
            {
                BeamRadius,
                PulseDuration,
                AveragePower,
                DutyCycle,
                PulseFrequency
            };
        }
    }
}

File: backend/model/laser.py

class Laser:
    """Plain old python object to hold laser properties."""
    def __init__(self, beam_radius, pulse_duration, avg_power, 
        duty_cycle, pulse_frequency):
        self.beam_radius = beam_radius
        self.pulse_duration = pulse_duration
        self.avg_power = avg_power
        self.duty_cycle = duty_cycle
        self.pulse_frequency = pulse_frequency
        self.pulse_average_power = avg_power / (pulse_frequency * pulse_duration)
